subprojects/core/src/integTest/groovy/org/gradle/internal/progress/BuildOperationServiceIntegrationTest.groovy

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.progress
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class BuildOperationServiceIntegrationTest extends AbstractIntegrationSpec {
+
+    def "build can listen to build operations events"() {
```

Comment:
I wouldn't necessarily add a functional test for this, or perhaps change it to show a plugin can use this service. 

Instead of a test, perhaps change the tooling API provider to use this service and remove `GradleLauncher.addNestedListener()` and `BuildController#addNestedListener()`. There's a couple of spots that would need to change.